unit_tests/freshclam_test.py

```diff
@@ -248,7 +247,7 @@ def test_freshclam_04_http_429(self):
         self.step_name('Verify correct behavior when receiving 429 (too-many-requests)')
 
         # Start our mock database mirror.
-        TC.mock_mirror = Process(target=mock_database_mirror, args=(WebServerHandler_04,TC.mock_mirror_port))
+        TC.mock_mirror_port, TC.mock_mirror = mock_database_mirror(WebServerHandler_04, TC.mock_mirror_port))
```

Comment:
I've run `ctest` and it passed, but I must have done something wrong, given there is this error!

Reply:
I'm not sure why this didn't fail for you.
It seems like the tests didn't run on this PR, or maybe aged off since it's been sitting so long.  I'll fix it for your and see if it triggers the CI pipeline (again?).

Reply:
BTW just confirmed manually that it fails because of this typo:
```

8: ______________________ ERROR collecting freshclam_test.py ______________________
8: ../../../.local/lib/python3.8/site-packages/_pytest/python.py:599: in _importtestmodule
8:     mod = import_path(self.path, mode=importmode, root=self.config.rootpath)
8: ../../../.local/lib/python3.8/site-packages/_pytest/pathlib.py:533: in import_path
8:     importlib.import_module(module_name)
8: /usr/lib/python3.8/importlib/__init__.py:127: in import_module
8:     return _bootstrap._gcd_import(name[level:], package, level)
8: <frozen importlib._bootstrap>:1014: in _gcd_import
8:     ???
8: <frozen importlib._bootstrap>:991: in _find_and_load
8:     ???
8: <frozen importlib._bootstrap>:975: in _find_and_load_unlocked
8:     ???
8: <frozen importlib._bootstrap>:671: in _load_unlocked
8:     ???
8: ../../../.local/lib/python3.8/site-packages/_pytest/assertion/rewrite.py:162: in exec_module
8:     source_stat, co = _rewrite_test(fn, self.config)
8: ../../../.local/lib/python3.8/site-packages/_pytest/assertion/rewrite.py:364: in _rewrite_test
8:     tree = ast.parse(source, filename=strfn)
8: /usr/lib/python3.8/ast.py:47: in parse
8:     return compile(source, filename, mode, flags,
8: E     File "/home/micasnyd/workspace/clamav-micah-3/unit_tests/freshclam_test.py", line 250
8: E       TC.mock_mirror_port, TC.mock_mirror = mock_database_mirror(WebServerHandler_04, TC.mock_mirror_port))
8: E
              ^
8: E   SyntaxError: unmatched ')'
8: =========================== short test summary info ============================
8: ERROR freshclam_test.py
8: !!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
8: =============================== 1 error in 0.33s ===============================
 8/11 Test  #8: freshclam ........................***Failed    0.83 sec
```

Reply:
Fixed it, and fixed a few other issues I found. 
I also rebased the branch so it is based off of the latest from `main`.